lxd/storage/drivers/generic_vfs.go

```diff
@@ -624,14 +624,32 @@ func genericVFSBackupVolume(d Driver, vol VolumeCopy, tarWriter *instancewriter.
 			snapshotsPrefix = "backup/volume-snapshots"
 		}
 
+		parentUUID := vol.config["volatile.uuid"]
+		if parentUUID == "" {
+			return fmt.Errorf(`Failed loading "volatile.uuid" from volume %q`, vol.Name())
+		}
+
 		for _, snapName := range snapshots {
-			prefix := filepath.Join(snapshotsPrefix, snapName)
-			snapVol, err := vol.NewSnapshot(snapName)
-			if err != nil {
-				return err
+			found := false
+			var snapVol Volume
+			for _, snapshot := range vol.Snapshots {
+				_, snapshotName, _ := api.GetParentAndSnapshotName(snapshot.name)
+				if snapshotName == snapName {
+					snapVol = snapshot
+					found = true
+					break
+				}
+			}
+
+			if !found {
+				return fmt.Errorf("Snapshot %q missing in volume's list", snapName)
 			}
 
-			err = backupVolume(snapVol, prefix)
+			// Set the parent volume's UUID.
+			snapVol.parentUUID = parentUUID
```

Comment:
Why not using setter function for this?

Reply:
Also can we keep the initialisation together by moving this into the loop where snapVol is assigned above?

Reply:
Sorry, I have missed this one, maybe something happened during rebase as the other one is using the setter. Thanks!